tests/common/test_cgroupapi.py

```diff
@@ -506,7 +506,42 @@ def mock_popen(*args, **kwargs):
                         self.assertEquals(extension_cgroups, [])
 
     @patch('time.sleep', side_effect=lambda _: mock_sleep(0.001))
-    def test_start_extension_command_should_use_fallback_option_if_systemd_times_out(self, _):
+    def test_start_extension_command_should_use_fallback_option_if_systemd_times_out_internally(self, _):
+        # Systemd has its own internal timeout which is not necessarily the same as the timeout we define for
+        # extension operations. When systemd times out, it will write that to stderr and exit with exit code 1.
+        # In that case, we will internally recognize the failure due to the non-zero exit code, not as a timeout.
+        original_popen = subprocess.Popen
+        systemd_timeout_command = "echo 'Failed to start transient scope unit: Connection timed out' >&2 && exit 1"
```

Comment:
Also I was going through the systemd fallback code and I noticed something, it doesnt matter if its a systemd timeout or a systemd execution failure (exit non-0), the exception is handled the same way. If that's the case then any particular reason why the tests are named 'systemd_times_out'?

Reply:
implementation vs behavior :)

though the implementation currently handles errors the same (whether they are time outs or not) we want to assert the behavior of important scenarios, timeout being one of them.

if the implementation changes, we want the behavior to remain the same.

Reply:
Ohh nice, I get the motivation behind writing the test. 
1 follow up though, if we're checking the behavior then shouldn't we be testing the behavior? Like for example this test has `exit 1` to fail the test which is not necessarily a systemd timeout. If we're validation a scenario then shouldn't that actually mock systemd timeout?

Reply:
Yup, absolutely. When systemd timeout it exits with 1, no stdout and a specific message in stderr. The mock replicates this behavior.

Reply:
Ahh very interesting, thanks for sharing! This is an interesting systemd behavior I was unaware of! :)

Reply:
One may argue that the mock should also sleep for the timeout period, but that would not fit with a unit tests, I think (too slow). An integration test may be worth considering.